exercises/binary-search-tree/binary_search_tree_test.rb

```diff
@@ -86,4 +86,31 @@ def test_iterating_over_complex_tree
     four.insert 5
     assert_equal [1, 2, 3, 4, 5, 6, 7], record_all_data(four)
   end
+
+  def test_each_returns_enumerator_if_no_block
+    skip
+
+    tree = Bst.new 4
+    [2, 1, 3, 6, 7, 5].each { |x| tree.insert x }
+    each_enumerator = tree.each
+
+    assert_kind_of Enumerator, each_enumerator
+
+    (1..7).each { |x| assert_equal(x, each_enumerator.next) }
+
+    assert_raises(StopIteration) { each_enumerator.next }
```

Comment:
@Cohen-Carlisle 
Interesting thought, but it's easy to slip up and have `#map` be called on the wrong object as `#each` typically returns the object itself when it does get a block. I haven't added a test for returning `self` as there are occasions where it requires choosing between exposing internals or adding extra level of indirection in the iteration. If you do return `self`, than `#map` doesn't actually test that its an enumerator.